guides/william/DNSNotes.md

```diff
@@ -0,0 +1,164 @@
+		- also used to create aliases of domain names
+		- same server can be accessesed through documents.example.com and docs.example.com because of CNAME
+	- _MX Record_: specifies a mail exchange server for a DNS domain name
+		- the information is used by Simple Mail Transfer Protocol (SMTP) to route emails to proper hosts
+	- _PTR Record_: (reverse of A and AAAA DNS Records) used to look up domain names based on IP addresses
+
+### Installation 
+- [Install Ubuntu](https://www.ubuntu.com/download/server)
+- Change network options to `Host-Only, DMZ`
+- Type in `ip link show` into terminal and you should see: lo, enp0s3
+	- `ip link show` : shows information for all interfaces 
+	- lo : loopback
+	- enp0s3 : virtual network driver
+- Add another interface
+	- type `sudo nano /etc/network/interfaces` to edit the network interfaces configuration file with console based text editor nano
+	- add 
+	```
+	auto enp0s3
+	iface enp0s3 inet static
+	address 172.20.240.23
+	netmask 255.255.255.0
+	gateway 172.20.240.254
+	dns-nameservers 8.8.8.8
+	```
+	- type `sudo service networking restart`
+	- verify netowrk interfaces with `ifconfig`
+	- verify connectivity with `ping 8.8.8.8` 
+		- `8.8.8.8` is google's DNS server
+
+
+### Configuration
+- [Helpful link for configuration process](https://www.ostechnix.com/install-and-configure-dns-server-ubuntu-16-04-lts/)
+- [Helpful link for understanding bind](http://www.firewall.cx/linux-knowledgebase-tutorials/system-and-network-services/829-linux-bind-introduction.html)
+- Update and install bind
+	- `sudo apt-get update`
+	- `sudo apt-get upgrade`
+	- `sudo apt-get install bind9 bind9utils bind9doc dnsutils`
+	- bind is a widely used domain name system software for your server to become a DNS for your network
+	- dnsutils is a package for testing and troubleshooting DNS related issues, including a tool named dig 
+- Configure caching name server
+	- `sudo nano /etc/bind/named.conf.options`
+	- uncomment and change the lines 
+	```
+	//forwarders {
+	//	0.0.0.0;
+	//};
+	```
+	to 
+	```
+	forwarders {
+		8.8.8.8;
+		8.8.4.4;
+	};
+	```
+	- Because of this, if our cached server fails to recognize a IP address, our server will forward the request to 8.8.8.8 and 8.8.4.4.
+		- 8.8.8.8 and 8.8.4.4 are both public Google DNS servers.  
+- Change dns-nameservers so that bind will handle it 
+	- `sudo nano /etc/network/interfaces`
+	- change to 
+	```
+	auto enp0s3
+	iface enp0s3 inet static
+		address 172.20.240.23
+		netmask 255.255.255.0
+		gateway 172.20.240.254
+		dns-nameservers 172.0.0.1
+	```
+- Refresh 
+	- `sudo ip addr flush enp0s3`
+		- `ip addr flush` removes all addresses for the interface `enp0s3`
+	- `sudo systemctl restart networking.service`
+		- `systemctl` is the system manager
+		- `restart networking.service` will restart the networking service that `systemctl` manages
+	- `sudo systemctl restart bind9`
+		- will restart bind9
+	- `ping www.google.com`
+		- verify internet connection

```

Comment:
Try to be more verbose in your guide. What should the person see to verify it worked? How do you interpret the results?